packages/ember-data/lib/system/model/model.js

```diff
@@ -777,7 +793,8 @@ var Model = Ember.Object.extend(Ember.Evented, {
     });
     //We use the pathway of setting the hasMany as if it came from the adapter
     //because the user told us that they know this relationships exists already
-    this._relationships[key].updateRecordsFromAdapter(recordsToSet);
+    this.get(key);
```

Comment:
I see I had thought the `this._relationships.get(key)` in the next line would also instantiate the relationship. My mistake.